### PR TITLE
Traffic source

### PR DESCRIPTION
**Pull request by akilananb**
Traffic source

### CODE REVIEW

File: server/transforms/ab-api.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ const getSession = (event, cookie) => {
 // Attempt to get anon session id from cookie
 const userIdPattern = /SIVISITOR=([^;]*)/;
 const getAnonymousSessionId = (cookie) => {
-	var matches = userIdPattern.exec(cookie) || [];
+	let matches = userIdPattern.exec(cookie) || [];
```
Comment (commuterjoy) [inline server/transforms/ab-api.js]: const?

File: server/transforms/traffic-source.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+
+module.exports = function (event) {
+
+	if (!process.env.transform_traffic_source && !process.env.mocha) {
+		return Promise.resolve({});
+	}
+
+	const segementFallback= () => {
+		const url = event.annotations().url;
```
Comment (commuterjoy) [inline server/transforms/traffic-source.js]: btw, I'm trying not to build too many dependencies on annotations as the pipeline will get slower the more annotations we chain together.

Comment (akilananb) [inline server/transforms/traffic-source.js]: Ya had it in mind! But based on scenario it depends on referrer information...Any way it has to wait for other first level annotation to complete, So used URL as well. 

Can we club referrer  and traffic source together...?

**Comment by markstephens** [comment]
:+1: can we merge it?

**Comment by commuterjoy** [comment]
Talking @markstephens it seems this API doesn't really work very well so we need to chat with someone on Monday to fix it. So I won't merge this for now.

**Comment by markstephens** [comment]
I think it was calling the wrong URL - needed an uppercase i when using segid. I've added a commit.

**Pull request merged**